packages/roosterjs-editor-plugins/lib/plugins/ImageResize/ImageResizeHandler.ts

```diff
@@ -0,0 +1,266 @@
+                          y == YCoordinate.Middle ? '50%' : '0px'
+                      }">
+                    <div style="position:relative;width:${HANDLE_SIZE}px;height:${HANDLE_SIZE}px;background-color: ${
+                          this.selectionBorderColor
+                      };cursor:${direction}-resize;${
+                          y == YCoordinate.Top ? 'top' : 'bottom'
+                      }:-${HANDLE_MARGIN}px;${
+                          x == XCoordinate.Left ? 'left' : 'right'
+                      }:-${HANDLE_MARGIN}px"></div></div>`
+                    : `<div style="position:absolute;left:0;right:0;top:0;bottom:0;border:solid 1px ${this.selectionBorderColor};pointer-events:none;">`;
+                const div = fromHtml(html, doc)[0] as HTMLElement;
+                wrapper.appendChild(div);
+
+                this.resizeHelpers[x][y] = direction
+                    ? new ResizeHelper(
+                          div,
+                          [x, y],
+                          this.onResizeBegin,
+                          this.onResize,
+                          this.onResizeEnd
+                      )
+                    : null;
+            }
+        }
+
+        // If the resizeDiv's image has a transform, apply it to the container
+        wrapper.style.transform = this.image.style.transform;
+        this.image.style.transform = '';
+        this.editor.select(wrapper, PositionType.After);
+    }
+
+    private onResizeBegin = () => {
+        if (this.image) {
+            this.editor.addUndoSnapshot();
+            this.initWidth = this.image.clientWidth;
+            this.initHeight = this.image.clientHeight;
+            this.ratio =
+                this.initWidth > 0 && this.initHeight > 0
+                    ? (this.initWidth * 1.0) / this.initHeight
+                    : 0;
+        }
+    };
+
+    private onResize = ([x, y]: ResizeContext, deltaX: number, deltaY: number, e: MouseEvent) => {
+        if (this.image) {
+            const horizontalOnly = x == XCoordinate.Center;
+            const verticalOnly = y == YCoordinate.Middle;
+            const shouldPreserveRatio =
+                !(horizontalOnly || verticalOnly) && (this.forcePreserveRatio || e.shiftKey);
+            let newWidth = horizontalOnly
+                ? this.initWidth
+                : Math.max(
+                      this.initWidth + deltaX * (x == XCoordinate.Left ? -1 : 1),
+                      this.minWidth
+                  );
+            let newHeight = verticalOnly
+                ? this.initHeight
+                : Math.max(
+                      this.initHeight + deltaY * (y == YCoordinate.Top ? -1 : 1),
+                      this.minHeight
+                  );
+
+            if (shouldPreserveRatio) {
+                newHeight = Math.min(newHeight, (newWidth * this.initHeight) / this.initWidth);
+                newWidth = Math.min(newWidth, (newHeight * this.initWidth) / this.initHeight);
```

Comment:
since this.ratio is calculated already, should this work:

newHeight = Math.min(newHeight, newWidth / this.ratio);
newWidth = Math.min(newWidth, newHeight * this.ratio);

Reply:
I think you are right